File: Solving-Foundations-of-Programming/src/main/Pairs.java
package main;

import java.util.HashMap;
import java.util.Map;

/*
 * https://codingbat.com/prob/p126332
 * 
 * Given an array of non-empty strings, create and return a Map<String, String> as follows: 
 * for each string add its first character as a key with its last character as the value.
 * 
 * pairs(["code", "bug"]) → {"b": "g", "c": "e"}
 * pairs(["man", "moon", "main"]) → {"m": "n"}
 * pairs(["man", "moon", "good", "night"]) → {"g": "d", "m": "n", "n": "t"}
 * */
public class Pairs {
	public Map<String, String> makePairs(String[] strings) {
		Map<String, String> result = new HashMap<String, String>();
		for (String str : strings) {
			String firstStr = str.substring(0, 1);
			String lastStr = str.substring(str.length() - 1);
			result.put(firstStr, lastStr);
		}
		return result;
	}
}

File: problem-solving-with-python/pairs.py
# https://codingbat.com/prob/p126332
#
# Given an array of non-empty strings, create and return a Map<String, String> as follows:
# for each string add its first character as a key with its last character as the value.
#
#
# pairs(["code", "bug"]) → {"b": "g", "c": "e"}
# pairs(["man", "moon", "main"]) → {"m": "n"}
# pairs(["man", "moon", "good", "night"]) → {"g": "d", "m": "n", "n": "t"}
def pairs(strings):
    result = {}
    for str in strings:
        firstStr = str[0]
        lastStr = len(str) - 1
        result[firstStr] = str[lastStr]
    return result

strings = ["man", "moon", "good", "night"]
result = pairs(strings)
print(result)